packages/puppeteer-core/src/util/decorators.ts

```diff
@@ -57,3 +59,26 @@ export function moveable<
   }
   return Class;
 }
+
+export function withIsolatedRealm<
```

Comment:
let's call it `runOnNewIsolatedHandle` ?

Reply:
`createFreshIsolatedHandle`, `withNewIsolatedHandle`, `withThisAsNewIsolatedHandle` or smth like this

Reply:
What about `usingIsolatedHandle`?

Reply:
how can we highlight that a different thisArg will used? I think `usingIsolatedHandle ` does not indicate this either. Perhaps `@bindIsolatedHandle`?

Reply:
Ah, bindIsolatedHandle is perfect!

Reply:
also, does not indicate to me that `this` would be different

Reply:
Hmm, FWIU, you want the name to reflect the expectation. Is the comment not sufficient?

Reply:
The comment is not sufficient IMO because it's not in visible at the usage site. `bindIsolatedHandle` sgtm